differ/mutationDiffer.go

```diff
@@ -90,20 +90,39 @@ type MutationDiffer struct {
 	utils           xdcrUtils.UtilsIface
 }
 
+type JsonBody struct {
+	Value    []byte
+	Cas      uint64
+	Datatype uint8
+}
+
+type JsonMeta struct {
+	Value    []byte
```

Comment:
You should embed `JsonBody` as part of the `JsonMeta` as a "body" type pointer
If the mode is both body and meta, then the body type pointer will have data
If it is just metadata, then the body will be nil indicating the jsonMeta is meta only

It'll be more intuitive when reading the code and also less duplicate

Reply:
As mentioned I now populate the data to be encoded to a map. These structures are no longer needed